SimG4Core/SensitiveDetector/interface/SensitiveDetectorMakerBase.h

```diff
@@ -47,8 +47,8 @@ class SensitiveDetectorMakerBase
 			const edm::ParameterSet& p,
 			const SimTrackManager* m,
 			SimActivityRegistry& reg,
-			std::auto_ptr<SensitiveTkDetector>& oTK,
-			std::auto_ptr<SensitiveCaloDetector>& oCalo) const =0;
+			std::unique_ptr<SensitiveTkDetector>& oTK,
+			std::unique_ptr<SensitiveCaloDetector>& oCalo) const =0;
```

Comment:
i have the same concern - so these functions do not keep this pointer around internally?